[RF] Test failure: TestLandauEvil.CompareFixedValuesNorm

### Check duplicate issues.

- [x] Checked for duplicates

### Description

Hi

I am getting a single test failure on a build of ROOT release version 6.34.02.  The failed test is 'TestLandauEvil.CompareFixedValuesNorm' in gtest-roofit-roofit-vectorisedPDFs-testLandau.  Please find the log below.

Cheers
Jake
```
262/1389 Testing: gtest-roofit-roofit-vectorisedPDFs-testLandau
262/1389 Test: gtest-roofit-roofit-vectorisedPDFs-testLandau
Command: "/gnu/store/50gia554cgc7c3ijb7f51yqbdrxq772r-cmake-minimal-3.24.2/bin/cmake" "-DCMD=/tmp/guix-build-root-6.34.02.drv-0/build/roofit/roofit/test/vectorisedPDFs/testLandau" "-DCWD=/tmp/guix-build-root-6.34.02.drv-0/build/roofit/roofit/test/vectorisedPDFs" "-DSYS=/tmp/guix-build-root-6.34.02.drv-0/build" "-P" "/tmp/guix-build-root-6.34.02.drv-0/source/cmake/modules/RootTestDriver.cmake"
Directory: /tmp/guix-build-root-6.34.02.drv-0/build/roofit/roofit/test/vectorisedPDFs
"gtest-roofit-roofit-vectorisedPDFs-testLandau" start time: Jan 28 08:44 UTC
Output:
----------------------------------------------------------
Running main() from /tmp/guix-build-googletest-1.12.1.drv-0/source/googletest/src/gtest_main.cc
[==========] Running 4 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 2 tests from TestLandauEvil
[ DISABLED ] TestLandauEvil.DISABLED_CompareFixedValuesUnnorm
[ RUN      ] TestLandauEvil.CompareFixedValuesNorm
Compare event 581	    1)  x = -61.175
  1) RooRealVar::  mean = 15.4951998032629
  2) RooRealVar:: sigma = 10.0827928564977

	scalar   = 3.0533256912989e-321	pdf->getVal() = 3.06814766067414e-320
	batch    = 3.05826634775732e-321
	rel diff = -0.00161550888529887
/tmp/guix-build-root-6.34.02.drv-0/source/roofit/roofit/test/vectorisedPDFs/VectorisedPDFTests.cxx:367: Failure
Expected equality of these values:
  nFarOff
    Which is: 1
  0u
    Which is: 0
Google Test trace:
/tmp/guix-build-root-6.34.02.drv-0/source/roofit/roofit/test/vectorisedPDFs/testLandau.cxx:46: Parameter set 3
	mean	15.4952
	sigma	10.0828
/tmp/guix-build-root-6.34.02.drv-0/source/roofit/roofit/test/vectorisedPDFs/VectorisedPDFTests.cxx:368: Failure
Expected: (sumDiffs.Sum() / outputsScalar.size()) > (-toleranceCompare), actual: -1.61551e-07 vs -5e-14
Batch outputs biased towards negative.
Google Test trace:
/tmp/guix-build-root-6.34.02.drv-0/source/roofit/roofit/test/vectorisedPDFs/testLandau.cxx:46: Parameter set 3
	mean	15.4952
	sigma	10.0828
/tmp/guix-build-root-6.34.02.drv-0/source/roofit/roofit/test/vectorisedPDFs/VectorisedPDFTests.cxx:370: Failure
Expected: (sqrt(sumVars.Sum() / outputsScalar.size())) < (toleranceCompare), actual: 1.61551e-05 vs 5e-14
High standard deviation for batch results vs scalar.
Google Test trace:
/tmp/guix-build-root-6.34.02.drv-0/source/roofit/roofit/test/vectorisedPDFs/testLandau.cxx:46: Parameter set 3
	mean	15.4952
	sigma	10.0828

Maximal relative error (scalar vs batch) is: 0.00161550888529887

[  FAILED  ] TestLandauEvil.CompareFixedValuesNorm (224 ms)
[ DISABLED ] TestLandauEvil.DISABLED_RunScalar
[ DISABLED ] TestLandauEvil.DISABLED_RunBatch
[ RUN      ] TestLandauEvil.CompareBatchScalar

Timer 'Fitting scalar mode Landau_evil':	0.150807s (CPU) 0.151264772s (wall)

Timer 'Fitting batch mode Landau_evil':	0.020896s (CPU) 0.020993528s (wall)
[       OK ] TestLandauEvil.CompareBatchScalar (174 ms)
[----------] 2 tests from TestLandauEvil (398 ms total)

[----------] 2 tests from TestLandau
[ DISABLED ] TestLandau.DISABLED_CompareFixedValuesUnnorm
[ RUN      ] TestLandau.CompareFixedValuesNorm

Maximal relative error (scalar vs batch) is: 1.41737160490179e-14

[       OK ] TestLandau.CompareFixedValuesNorm (10 ms)
[ DISABLED ] TestLandau.DISABLED_CompareFixedNormLog
[ DISABLED ] TestLandau.DISABLED_RunScalar
[ DISABLED ] TestLandau.DISABLED_RunBatch
[ RUN      ] TestLandau.CompareBatchScalar

Timer 'Fitting scalar mode Landau_convenient':	0.041792s (CPU) 0.041867778s (wall)

Timer 'Fitting batch mode Landau_convenient':	0.010536s (CPU) 0.010537136s (wall)
[       OK ] TestLandau.CompareBatchScalar (53 ms)
[----------] 2 tests from TestLandau (64 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 2 test suites ran. (462 ms total)
[  PASSED  ] 3 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestLandauEvil.CompareFixedValuesNorm

 1 FAILED TEST
  YOU HAVE 7 DISABLED TESTS

CMake Error at /tmp/guix-build-root-6.34.02.drv-0/source/cmake/modules/RootTestDriver.cmake:232 (message):
  error code: 1


<end of output>
Test time =   0.53 sec
----------------------------------------------------------
Test Failed.
"gtest-roofit-roofit-vectorisedPDFs-testLandau" end time: Jan 28 08:44 UTC
"gtest-roofit-roofit-vectorisedPDFs-testLandau" time elapsed: 00:00:00
----------------------------------------------------------
```

### Reproducer

ctest -j N

or more specifically:

/gnu/store/50gia554cgc7c3ijb7f51yqbdrxq772r-cmake-minimal-3.24.2/bin/cmake -DCMD=/tmp/guix-build-root-6.34.02.drv-0/build/roofit/roofit/test/vectorisedPDFs/testLandau -DCWD=/tmp/guix-build-root-6.34.02.drv-0/build/roofit/roofit/test/vectorisedPDFs -DSYS=/tmp/guix-build-root-6.34.02.drv-0/build -P /tmp/guix-build-root-6.34.02.drv-0/source/cmake/modules/RootTestDriver.cmake

### ROOT version

  | Welcome to ROOT 6.34.02                        https://root.cern |
  | (c) 1995-2024, The ROOT Team; conception: R. Brun, F. Rademakers |
  | Built for linuxx8664gcc on Jan 01 1970, 00:00:01                 |
  | From tags/6-34-02@6-34-02                                        |
  | With c++ (GCC) 11.4.0                                            |
  | Try '.help'/'.?', '.demo', '.license', '.credits', '.quit'/'.q'  |


### Installation method

Build from source

### Operating system

Linux

### Additional context

ROOT was built from source with support for the following features:

-- Enabled support for:  asimage builtin_clang builtin_cling builtin_llvm builtin_nlohmannjson builtin_openui5 dataframe fitsio gdml geom http imt mysql opengl pgsql pyroot roofit root7 rpath runtime_cxxmodules shared spectrum sqlite ssl testsupport tmva tmva-cpu tmva-cudnn tmva-pymva tpython webgui x11 xml xrootd

--- a/math/mathcore/src/PdfFuncMathCore.cxx
+++ b/math/mathcore/src/PdfFuncMathCore.cxx
@@ -46,16 +46,14 @@ namespace Math {
 
       if (xi <= 0) return 0;
       double v = (x - x0)/xi;
-      double u, ue, us, denlan;
+      double u, denlan;
       if (v < -5.5) {
          u   = std::exp(v+1.0);
          if (u < 1e-10) return 0.0;
-         ue  = std::exp(-1/u);
-         us  = std::sqrt(u);
-         denlan = 0.3989422803*(ue/us)*(1+(a1[0]+(a1[1]+a1[2]*u)*u)*u);
+         denlan = 0.3989422803*std::exp(-1 / u - 0.5 * (v + 1))*(1+(a1[0]+(a1[1]+a1[2]*u)*u)*u);
       } else if(v < -1) {
          u   = std::exp(-v-1);
-         denlan = std::exp(-u)*std::sqrt(u)*
+         denlan = std::exp(-u - 0.5 * (v + 1))*
             (p1[0]+(p1[1]+(p1[2]+(p1[3]+p1[4]*v)*v)*v)*v)/
             (q1[0]+(q1[1]+(q1[2]+(q1[3]+q1[4]*v)*v)*v)*v);
       } else if(v < 1) {

